odoo/addons/base/wizard/base_partner_merge.py

```diff
@@ -324,9 +324,16 @@ def _merge(self, partner_ids, dst_partner=None, extra_checks=True):
         _logger.info("dst_partner: %s", dst_partner.id)
 
         # FIXME: is it still required to make and exception for account.move.line since accounting v9.0 ?
-        if extra_checks and 'account.move.line' in self.env and self.env['account.move.line'].sudo().search([('partner_id', 'in', [partner.id for partner in src_partners])]):
+        if extra_checks and 'account.move.line' in self.env and self.env['account.move.line'].sudo().search([('partner_id', 'in', [partner.id for partner in src_partners])], limit=1):
             raise UserError(_("Only the destination contact may be linked to existing Journal Items. Please ask the Administrator if you need to merge several contacts linked to existing Journal Items."))
 
+        # Prevent break the hash
+        if 'account.move.line' in self.env and self.env['account.move.line'].sudo().search([
+            ('partner_id', 'in', [partner.id for partner in src_partners]),
+            ('parent_state', '=', 'posted'),
+            ('move_id.secure_sequence_number', '!=', 0)], limit=1):
+            raise UserError(_('You cannot merge a contact linked to a posted and hashed Journal Entry with another contact.'))
```

Comment:
```suggestion
        if 'account.move.line' in self.env and self.env['account.move.line'].sudo().search([
            ('partner_id', 'in', src_partners.ids),
            ('parent_state', '=', 'posted'),
            ('move_id.secure_sequence_number', '!=', 0)], limit=1):
            raise UserError(_('You cannot merge a contact linked to a posted and hashed Journal Entry with another contact.'))
```
Don't forget to add the new translated term in the .pot file of the module.

I don't think checking if 'account.move.line' is in the env is useful at all, but as I see it done above I'll check with someone else to understand better why it is like that and come back to you later.